### PR TITLE
Fixes exact checkbox alignment and backend logic

### LINKED ISSUE #3675: Search / Criteria"Exact" overwrites 'Add" button
The search is a great feature (quick to fill, powerfull  - with advanced criterias - and results are fast and reliable) but I have noticed an issue in the layout : "Exact" (which seems to be an untranslatable string yet) overwrite the "Add" button (at least in french, the full french label is "Ajouter"); see:  Search

page https://hosted.weblate.org/projects/weblate/master/#search

### PR DESCRIPTION
**Pull request by SaptakS**
Before submitting pull request, please ensure that:

- [x] Every commit has message which describes it
- [x] Every commit is needed on it's own, if you have just minor fixes to previous commits, you can squash them
- [x] Any code changes come with testcase
- [x] Any new functionality is covered by user documentation

Fixes #3675 

- Used `.input-group-addon` for the exact checkbox instead of an absolute label.
- Fixed the regex expression to match quotes for multi-word expression and without(or with) quotes for single word.
- Added documentation for the same.

### CODE REVIEW

File: weblate/utils/search.py [hunk 1]
```diff
@@ -69,16 +69,16 @@ class Exact(whoosh.query.Term):
...
+    """Exact match plugin to specify an exact term."""
 
     class ExactNode(whoosh.qparser.syntax.TextNode):
         qclass = Exact
 
         def r(self):
             return "Exact %r" % self.text
 
-    expr = r"\=(^|(?<=\W))['\"](?P<text>.*?)['\"](?=\s|\]|[)}]|$)"
+    expr = r"\=(^|(?<=\W))(['\"]?)(?P<text>.*?)\2(?=\s|\]|[)}]|$)"
```
Comment (nijel) [inline weblate/utils/search.py:81]: This seems to break the tests: `AssertionError: <Q: (AND: ('source__iexact', 'hello'))> != <Q: (AND: ('source__substring', '=hello'))>`, see https://github.com/WeblateOrg/weblate/pull/3677/checks?check_run_id=563709490#step:8:575

Comment (SaptakS) [reply weblate/utils/search.py:81]: Yeah. I guess it should be equal. :thinking:

Comment (SaptakS) [reply weblate/utils/search.py:81]: Previously, quotes were needed, so I added that test.

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**